app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  has_many :votes, dependent: :destroy
+
+  validates :username, presence: {message: "Username is required."}
+  validates :username, uniqueness: {message: "Username already exists."}
+
+
+
+  def list_votes(user)
+    @votes = Vote.all
+    @works = Work.all
+    user_works = []
+    votes = @votes.where(user_id: user)
```

Comment:
I really like that you've implemented this functionality in the model.

However, I'm a little worried about this implementation - you should be using active record relations to do some of this work for you. For example, instead of calling `Vote.all`, you can get a list of a user's votes by saying `user.votes`, and instead of saying `@works.find(vote.work_id)`, you can say `vote.work`.

Also, since this is already an instance method on `User`, you shouldn't need to pass in a user, you can just use `self`.

Here is a simplified version:

```ruby
def list_votes
  user_works = []
  self.votes.each do |vote|
    user_works << vote.work.title
  end
  return user_works
end
```